plugins/carbon_black_cloud/icon_carbon_black_cloud/connection/connection.py

```diff
@@ -146,3 +152,77 @@ def test(self):
                 data=str(error),
             )
         return {"success": True}
+
+    def test_task(self):
+        self.logger.info("Running a connection test to Carbon Black Cloud")
+
+        alerts_endpoint = self.base_url + f"/api/alerts/v7/orgs/{self.org_key}/alerts/_search"
+        observations_endpoint = self.base_url + f"/api/investigate/v2/orgs/{self.org_key}/observations/search_jobs"
+
+        current_time = datetime.now(timezone.utc)
+        now = current_time.strftime(TIME_FORMAT)
+        minus_five_mins = (current_time - timedelta(minutes=5)).strftime(TIME_FORMAT)
+
+        fd = {
+            alerts_endpoint: {
+                "name": "alerts",
+                "params": {
+                    "time_range": {"start": minus_five_mins, "end": now},
+                    "criteria": {},
+                    "start": "1",
+                    "rows": str(1),
+                    "sort": [{"field": ALERT_TIME_FIELD, "order": "ASC"}],
+                },
+            },
+            observations_endpoint: {
+                "name": "observations",
+                "params": {
+                    "rows": 1,
+                    "start": 0,
+                    "fields": ["*"],
+                    "criteria": {"observation_type": OBSERVATION_TYPES},
+                    "sort": [{"field": OBSERVATION_TIME_FIELD, "order": "asc"}],
+                    "time_range": {"start": minus_five_mins, "end": now},
+                },
+            },
+        }
+
+        return_msg = "Task connection test to Carbon Black Cloud successful"
+        return_msg_error = "Task connection test to Carbon Black Cloud failed"
+
+        failed = ""
+        failed_endpoint = []
+
+        for key, value in fd.items():
+            self.logger.info(f"Testing {value.get('name')} endpoint")
+
+            try:
+                self.request_api(key, value.get("params"))
+
+            except HTTPErrorException as error:
```

Comment:
lgtm! Just one final ask is if we can include a few loggers in these excepts for traceability in insightops to track each failure type and if the connection test was successful?